src/internal/WixBuildTools.TestSupport/WixAssert.cs

```diff
@@ -91,5 +91,54 @@ public int GetHashCode(object obj)
                 return this.stringComparer.GetHashCode((string)obj);
             }
         }
+
+        // xxxxx There is a bug in Xunit: It does not pass the System.Threading.Tasks.Task
+        // xxxxx metadata to Xunit's clients, even though it exposes the entrypoint. C++/CLR clients
+        // xxxxx are therefore deprived of the information that they need to use this entrypoint.
+        // xxxxx C# clients, however, are not dependent on this metadata, at least not for this purpose.
+        // xxxxx
+        // xxxxx All such types referenced by Xunit's own metadata are treated as <error-type>,
+        // xxxxx making them indistinguishable from each other and, in the case of C++/CLR, this type
+        // xxxxx is indistinguishable from System.Object.
+        // xxxxx
+        // xxxxx When C++/CLI tries to perform overload resolution, it can't distinguish between
+        // xxxxx System.Threading.Tasks.Task and System.Object, making the code ambiguous. The compiler
+        // xxxxx either reports an inscrutable error message or no error at all. It also doesn't emit
+        // xxxxx any code, so execution falls through and it appears that the test has passed, even
+        // xxxxx though it hasn't even been executed.
+        // xxxxx
+        // xxxxx The following method is used to isolate DUtilTests in order to overcome the above problem.
+
+        /// <summary>
+        /// This shim allows C++/CLR code to call the Xunit method with the same signature
+        /// without getting an ambiguous overload error.  If the specified test code
+        /// fails to generate an exception of the exact specified type, an assertion
+        /// exception is thrown. Otherwise, execution flow proceeds as normal.
+        /// </summary>
+        /// <typeparam name="T">The type name of the expected exception.</typeparam>
+        /// <param name="testCode">An Action delegate to run the test code.</param>
+        public static new void Throws<T>(System.Action testCode)
+            where T : System.Exception
+        {
+            Xunit.Assert.Throws<T>(testCode);
```

Comment:
If you want to be explicit with `Xunit.Assert`, then I would move these into a separate class. That would also make it possible to not need the `new` keyword to avoid hiding the base class's version. I'm actually a little surprised that the C++/CLI can handle the `new` functionality of hiding the other one if it can't handle the overloading problem.